Stencil/Node.swift

```diff
@@ -118,33 +117,37 @@ public class ForNode : NodeType {
   public class func parse(parser:TokenParser, token:Token) throws -> NodeType {
     let components = token.components()
 
-    if components.count == 4 && components[2] == "in" {
-      let loopVariable = components[1]
-      let variable = components[3]
-
-      var emptyNodes = [NodeType]()
-
-      let forNodes = try parser.parse(until(["endfor", "empty"]))
-
-      if let token = parser.nextToken() {
-        if token.contents == "empty" {
-          emptyNodes = try parser.parse(until(["endfor"]))
-          parser.nextToken()
-        }
-      } else {
-        throw TemplateSyntaxError("`endfor` was not found.")
-      }
+    guard components.count == 4 && components[2] == "in" else {
+      throw TemplateSyntaxError("'for' statements should use the following 'for x in y' `\(token.contents)`.")
+    }
+    
+    let loopVariable = components[1]
+    let variable = components[3]
+    
+    var emptyNodes = [NodeType]()
+    
+    parser.eatNextCRLF() // If CRLF after {% for %}, eat it
+    let forNodes = try parser.parse(until(["endfor", "empty"]))
+    
+    guard let token = parser.nextToken() else {
+      throw TemplateSyntaxError("`endfor` was not found.")
+    }
 
-      return ForNode(variable: variable, loopVariable: loopVariable, nodes: forNodes, emptyNodes:emptyNodes)
+    if token.contents == "empty" {
+      parser.eatNextCRLF() // if CRLF after {% empty %}, eat it
+      emptyNodes = try parser.parse(until(["endfor"]))
+      parser.nextToken()
     }
+    parser.eatNextCRLF() // if CRLF after {% endfor %}, eat it
 
-    throw TemplateSyntaxError("'for' statements should use the following 'for x in y' `\(token.contents)`.")
+    return ForNode(variable: variable, loopVariable: loopVariable, nodes: forNodes, emptyNodes: emptyNodes)
   }
 
   public init(variable:String, loopVariable:String, nodes:[NodeType], emptyNodes:[NodeType]) {
     self.variable = Variable(variable)
     self.loopVariable = loopVariable
     self.nodes = nodes
+    // TODO: Handle emptyNodes
```

Comment:
I wonder why the compiler didn't warn about unused variable.parameter btw, I'd expect it to suggest you a _Fix-It_ to replace `emptyNodes` with `_` in such cases.